odoo/service/server.py

```diff
@@ -372,20 +376,24 @@ def process_limit(self):
             self.limits_reached_threads.add(threading.currentThread())
 
         for thread in threading.enumerate():
-            if not thread.daemon or getattr(thread, 'type', None) == 'cron':
-                # We apply the limits on cron threads and HTTP requests,
-                # longpolling requests excluded.
-                if getattr(thread, 'start_time', None):
-                    thread_execution_time = time.time() - thread.start_time
-                    thread_limit_time_real = config['limit_time_real']
-                    if (getattr(thread, 'type', None) == 'cron' and
-                            config['limit_time_real_cron'] and config['limit_time_real_cron'] > 0):
-                        thread_limit_time_real = config['limit_time_real_cron']
-                    if thread_limit_time_real and thread_execution_time > thread_limit_time_real:
-                        _logger.warning(
-                            'Thread %s virtual real time limit (%d/%ds) reached.',
-                            thread, thread_execution_time, thread_limit_time_real)
-                        self.limits_reached_threads.add(thread)
+            # We apply the limits on cron threads and HTTP requests,
+            # longpolling requests excluded.
+            if (thread.daemon
+                and getattr(thread, 'type', None) not in ('cron', 'async')
+                or not getattr(thread, 'start_time', None)):
+                continue
+
+            execution_time = time.time() - thread.start_time
+            limit_time = config['limit_time_real']
+            if config.get('limit_time_real_%s' % thread.type, -1) != -1:
+                limit_time = config["limit_time_real_%s" % thread.type]
+
+            if limit_time and execution_time > limit_time:
+                _logger.warning(
+                    'Thread %s virtual real time limit (%d/%ds) reached.',
+                    thread, execution_time, limit_time)
+                self.limits_reached_threads.add(thread)
```

Comment:
I'm not saying the original code is great, but it could have been updated as-is for a much smaller and simpler diff.

Reply:
IMO if I try the minimal diff strategy on that one, the code would be even more dense. Harder diff now, IMO simpler function then

Reply:
```python
if not thread.daemon or getattr(thread, 'type', None) in ('cron', 'async'):
    if getattr(thread, 'start_time', None):
        thread_execution_time = time.time() - thread.start_time
        thread_limit_time_real = config['limit_time_real']
        if config.get('limit_time_real_%s' % thread.type, 0) > 0:
            thread_limit_time_real = config['limit_time_real_%s' % thread.type]
        if thread_limit_time_real and thread_execution_time > thread_limit_time_real:
            _logger.warning(
                'Thread %s virtual real time limit (%d/%ds) reached.',
                thread, thread_execution_time, thread_limit_time_real)
            self.limits_reached_threads.add(thread)
```
looks about fine to me, and has the advantage of changing very little from the original.

Reply:
What about the magic 0 value ? From --help `Set to 0 for no limit.`